uber/site_sections/preregistration.py

```diff
@@ -98,6 +98,48 @@ def index(self, message=''):
     def dealer_registration(self, message=''):
         return self.form(badge_type=c.PSEUDO_DEALER_BADGE, message=message)
 
+    def redeem(self, session, message='', **params):
```

Comment:
I'm just gonna chime in here and say that as Joe Rando, I'd be pretty
pissed if:

   1. I filled out the form
   2. I entered my coupon code
   3. My coupon code was validated and showed my discount
   4. I pull out my ... dammit, where's my wallet?
   5. ...time passes (5 minutes, an hour? doesn't matter)
   6. I try to pay and am told either:
      1. Total charged doesn't reflect the coupon discount
      2. OR "Hey, your total changed, do you still want to charge?"

IMO we *really* need to avoid the code running out while they're looking
for their CC.

Note that I'm NOT talking about them coming back days later.

If there's a time expiration, that should pop up right @ the code box
warning them how much time they have.
If there's a usage # limit, ditto.

And tho I didn't speak up before, nearly every single thing I buy on the
'Net has a "Coupon Code" box. If I care enough to try, I might spend five
minutes looking for a code. If I don't find one, or one I find doesn't
work, I don't get pissed, and it is incredibly unlikely to change my
purchase plans at that point in time, or in the future. I vote for a
visible, *easy-to-locate* code box. I *hate* searching for one.

--
Robert Earl <r.a.earl@gmail.com>
FAX: 1-440-919-5086

On Sat, Aug 6, 2016 at 10:30 PM, Eli Courtwright <notifications@github.com>
wrote:

> In uber/site_sections/preregistration.py
> <https://github.com/magfest/ubersystem/pull/1918#discussion_r73794790>:
>
> > @@ -98,6 +98,48 @@ def index(self, message=''):
> >      def dealer_registration(self, message=''):
> >          return self.form(badge_type=c.PSEUDO_DEALER_BADGE, message=message)
> >
> > +    def redeem(self, session, message='', **params):
>
> To build on what I said earlier and what @kitsuta
> <https://github.com/kitsuta> has also said, I think there's a different
> approach we should take here.
>
> The current workflow for the attendee is currently:
>
>    - Attendee fills out the reg form.
>    - Before they pay, they fill out the "redeem coupon" form.
>    - At the time they fill out that form, the validity is checked.
>    - Attendee then pays.
>
> There are some problems with this approach. A better approach would
> probably be:
>
>    - Attendee fills out the prereg form. During this time, they can enter
>    coupon codes into the "Comments" field.
>    - A discount is applied based on that coupon code. However, the coupon
>    is not actually redeemed until they try to pay. It's validity is checked
>    both at that time and when the payment goes through and the attendee is
>    saved.
>
> This implementation would be a little trickier, but I think it would be
> better in the long run. With that being said, I'm happy to defer to Dom on
> this, who's working more closely with you on this, and who has spent more
> time reviewing this.
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/magfest/ubersystem/pull/1918/files/4d15ba1149274f06e059b696d63509fee20b7aec#r73794790>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AIXSFRcqQl0wgqu5Fg7TaOHpA_rp1iuxks5qdUMogaJpZM4JcD4R>
> .
>